### PR TITLE
voxcpm2: fix inference correctness, add Metal GPU support and continuation voice cloning

### PR DESCRIPTION
**Pull request by ZMXJJ**
## Overview

This PR addresses three areas of the VoxCPM2 TTS pipeline:

1. **Inference correctness** — Fixed two numerical bugs that caused audio drift: MiniCPM4 was using NORM RoPE but the weights expect NEOX-style rotate_half; moved MINICPM4 to the NEOX group so C++ hidden states match PyTorch exactly. ResidualLM prefill ran with an uninitialized softmax mask (leaf tensor was never found by the op-node scan), corrupting prefill output and KV-cache. Rebuilt the mask as an explicit ggml_set_input and unified the residual_hidden path.

2. **Metal GPU end-to-end** — VoxCPM2 previously fell back to CPU because the Metal PAD kernel rejects left-padding. Replaced the AudioVAE left-pad with an equivalent ggml_concat of a zero block. Also fixed the residency-set teardown assertion (static destruction ordering) and added std::_Exit() for clean process exit. RTF improved from ~2.56 (CPU) to ~1.94 (Metal).

3. **Continuation voice cloning** — Wired the previously-dead --prompt-wav / --prompt-text CLI flags to a working continuation conditioning path, reusing the existing prefill/prefix_feat_cond graphs.

Additionally, a CMake smoke-test target was added for voxcpm2-cli.

## Additional information

- RoPE change is scoped to LLM_ARCH_MINICPM4 globally; correct for VoxCPM2 BaseLM (no Q/K permute at convert time). Other MINICPM4 GGUFs should be verified independently.
- Metal fix touches only voxcpm2_audiovae.cpp (ggml core untouched) and ggml-metal-device.m (teardown tolerance only).
- Voice cloning does not modify upstream/core files; zero-shot, reference, and streaming paths are unchanged.

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: YES — AI assisted with code review and debugging. All changes have been verified by me and through test cases.

### CODE REVIEW

File: src/llama-model.cpp [hunk 1]
```diff
@@ -2289,7 +2289,6 @@ llama_rope_type llama_model_rope_type(const llama_model * model) {
         case LLM_ARCH_STARCODER:
         case LLM_ARCH_INTERNLM2:
         case LLM_ARCH_MINICPM:
-        case LLM_ARCH_MINICPM4:
```
Comment (tc-mb) [inline src/llama-model.cpp:2292]: @ZMXJJ 这个好像是只是修改个位置，看着没意义，感觉可以不改。

**Comment by tc-mb** [comment]
@ZMXJJ 希望可以清理一下cursor等ai的签名。

File: src/llama-model.cpp (hunk 1 above)
Comment (oreomaker) [reply src/llama-model.cpp:2292]: MiniCPM4使用的是NEOX类型，这个在之前合并上游代码的pr中进行了更正，但可能已经被覆盖，确实需要进行这个更改。

Comment (tc-mb) [reply src/llama-model.cpp:2292]: @oreomaker 他好像就换了个位置？

Comment (oreomaker) [reply src/llama-model.cpp:2292]: 、NORM的RoPE和pytorch模型NOEX实现的差异会让音频生成完全不对，所以MINICPM4应该是放到NEOX里

Comment (ZMXJJ) [reply src/llama-model.cpp:2292]: @tc-mb 这里是吧MINICPM4 从 LLAMA_ROPE_TYPE_NORM 组移到了 LLAMA_ROPE_TYPE_NEOX 组

Comment (tc-mb) [reply src/llama-model.cpp:2292]: ok，看了下代码，get

**Comment by ZMXJJ** [comment]
> @ZMXJJ 希望可以清理一下cursor等ai的签名。

@tc-mb done
